backend/note_taking/serializers/block_list_serializer.py

```
@@ -22,37 +23,77 @@ def create(self, validated_data):
 
                 relative_position = item.pop("relative_position", None)
                 following_blocks = item.pop("following_blocks", None)
+                
+                if following_blocks:
+                    last_following_block = following_blocks[-1]
+                else:
+                    last_following_block = None
```

Ruff (0.13.1)

Local variable `last_following_block` is assigned to but never used

Remove assignment to unused variable `last_following_block`

(F841)